proxy_presentation.php

```diff
@@ -60,14 +60,37 @@
     }
 
     $bbbinstanceid = $meetingidparts[2];
+    $basemeetingid = $meetingidparts[0];
 
     // MeetingID can sometimes have [xxx] at the end. Needs sanitisation.
     if (strpos($bbbinstanceid, '[') !== false) {
         $bbbinstanceid = substr($bbbinstanceid, 0, strpos($bbbinstanceid, '['));
     }
 
-    $viewinstance = bigbluebuttonbn_view_instance_bigbluebuttonbn($bbbinstanceid);
-    require_login($viewinstance['course'], true, $viewinstance['cm']);
+
+    // Check if the user has appropriate access to the meeting resources.
+    $meetingsinstances = $DB->get_records('bigbluebuttonbn', ['meetingid' => $basemeetingid], '', 'id, course');
+    $access = false;
+    $counter = 1;
+    $maxcount = count($meetingsinstances);
+    foreach ($meetingsinstances as $bbbinstance) {
+        try {
+            $viewinstance = bigbluebuttonbn_view_instance_bigbluebuttonbn($bbbinstance->id);
```

Comment:
Would it be more performant to pull just the required metadata straight from the DB?

Reply:
That BBB helper function does just that (albeit in several small queries)
```
function bigbluebuttonbn_view_instance_bigbluebuttonbn($bigbluebuttonbnid) {
    global $DB;
    $bigbluebuttonbn = $DB->get_record('bigbluebuttonbn', array('id' => $bigbluebuttonbnid), '*', MUST_EXIST);
    $course = $DB->get_record('course', array('id' => $bigbluebuttonbn->course), '*', MUST_EXIST);
    $cm = get_coursemodule_from_instance('bigbluebuttonbn', $bigbluebuttonbn->id, $course->id, false, MUST_EXIST);
    return array('cm' => $cm, 'course' => $course, 'bigbluebuttonbn' => $bigbluebuttonbn);
}
```